R/makephyloseqFromTreeSummarizedExperiment.R

```diff
@@ -0,0 +1,113 @@
+#' Create a phyloseq object from a TreeSummarizedExperiment object
+#'
+#' @param x a \code{TreeSummarizedExperiment} object
+#'
+#' @param abund_values
+#' A single character value for selecting the
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}}
+#'   to be included in the phyloseq object that is created.
+#'
+#'  @details
+#' \code{makePhyloseqFromTreeSummarizedExperiment} is used for creating a phyloseq
+#' object from TreeSummarizedExperiment object.
+#'
+#' @return
+#' An object of class \code{Phyloseq} object.
+#'
+#' @name makePhyloseqFromTreeSummarizedExperiment
+#' @export
+#'
+#' @author Leo Lahti and Tuomas Borman. Contact: \url{microbiome.github.io}
+#'
+#' @examples
+#' # Get TSE object
+#' data("GlobalPatterns")
+#' TSE <- GlobalPatterns
+#'
+#' # Create a phyloseq object from it
+#' phy <- makePhyloseqFromTreeSummarizedExperiment(TSE)
+#'
+#' # By default the chosen table is counts, but if there is other tables,
+#' # they can be chosen with abund_values.
+#'
+#' # Counts relative abundances table
+#' TSE <- transformCounts(TSE, method = "relabundance")
+#' phy2 <- makePhyloseqFromTreeSummarizedExperiment(TSE, abund_values = "relabundance")
+#'
+NULL
+
+#' @rdname makePhyloseqFromTreeSummarizedExperiment
+#' @export
+setGeneric("makePhyloseqFromTreeSummarizedExperiment", signature = c("x"),
+           function(x, abund_values = "counts")
+               standardGeneric("makePhyloseqFromTreeSummarizedExperiment"))
+
+
+#' @rdname makePhyloseqFromTreeSummarizedExperiment
+#' @export
+setMethod("makePhyloseqFromTreeSummarizedExperiment", signature = c(x = "SummarizedExperiment"),
+          function(x, abund_values = "counts"){
+
+            # Input check
+            # Check object
+            if(!is(x,"TreeSummarizedExperiment")){
+                stop("'x' must be a 'TreeSummarizedExperiment' object")
+            }
```

Comment:
This has to be restructured. Since you now, which type of object your are handling, you can handle all different `SE` classes in a nested way.

```
setGeneric("makePhyloseqFromTreeSummarizedExperiment", signature = c("x"),
            function(x, ...)
                standardGeneric("makePhyloseqFromTreeSummarizedExperiment"))

setMethod("makePhyloseqFromTreeSummarizedExperiment", signature = c(x = "SummarizedExperiment"),
           function(x, abund_values = "counts"){
        # do conversion to all the counts data, sample data and feature data
    }
)

setMethod("makePhyloseqFromTreeSummarizedExperiment", signature = c(x = "TreeSummarizedExperiment"),
           function(x, ...){
        ans <- callNextMethod()
        # add tree data if present, add sequence data if present
        ans
    }
)
```

Reply:
Restructured